mautic/mautic
# [UX] Allow resource type selection using URL parameter

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | 🔴 
| New feature/enhancement? (use the a.x branch)      | 🟢
| Deprecations?                          | 🔴
| BC breaks? (use the c.x branch)        | 🔴
| Automated tests included?              | 🔴 
| Related user documentation PR URL      | mautic/user-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation-new#... 
| Issue(s) addressed                     | Fixes #... 



## Description

This PR allows devs to make links that automatically select a form or email type using an URL parameter instead of prompting the user for this.

URL examples:
`/s/emails/new?emailType=template` will automatically create a new email selecting the template type 
`/s/forms/new?formType=campaign` will create a campaign form

Links example:

```
<a href="{{ path('mautic_form_action', {'objectAction': 'new', 'formType': 'campaign'}) }}">Create Campaign Form</a>
<a href="{{ path('mautic_form_action', {'objectAction': 'new', 'formType': 'standalone'}) }}">Create Standalone Form</a>

<a href="{{ path('mautic_email_action', {'objectAction': 'new', 'emailType': 'template'}) }}">Create Template Email</a>
<a href="{{ path('mautic_email_action', {'objectAction': 'new', 'emailType': 'list'}) }}">Create Segment Email</a>
```







---
### 📋 Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Append `/s/emails/new?emailType=template` to your Mautic URL. Example (if testing locally): https://127.0.0.1:32769/s/emails/new?emailType=template 
3. Do the same for /s/forms/new?formType=campaign
4. Fill the info and save
5. Go back to the list
6. You will see that the created form has the Campaign form icon and the email has no segment icon
